Fix build with OpenSSL 4.x

Downloaded from
https://github.com/botovq/ports/commit/bb50119f0a15fdda547ffbf1e8d947d92470ed4e

Upstream: n/a (QT 5.x is not updated anymore)

Signed-off-by: Bernd Kuhls <bernd@kuhls.net>

Index: src/network/ssl/qsslcertificate_openssl.cpp
--- a/src/network/ssl/qsslcertificate_openssl.cpp.orig
+++ b/src/network/ssl/qsslcertificate_openssl.cpp
@@ -123,11 +123,11 @@ QByteArray QSslCertificate::serialNumber() const
     QMutexLocker lock(QMutexPool::globalInstanceGet(d.data()));
 #endif
     if (d->serialNumberString.isEmpty() && d->x509) {
-        ASN1_INTEGER *serialNumber = q_X509_get_serialNumber(d->x509);
+        const ASN1_INTEGER *serialNumber = q_X509_get_serialNumber(d->x509);
         QByteArray hexString;
-        hexString.reserve(serialNumber->length * 3);
-        for (int a = 0; a < serialNumber->length; ++a) {
-            hexString += QByteArray::number(serialNumber->data[a], 16).rightJustified(2, '0');
+        hexString.reserve(q_ASN1_STRING_length(serialNumber) * 3);
+        for (int a = 0; a < q_ASN1_STRING_length(serialNumber); ++a) {
+            hexString += QByteArray::number(q_ASN1_STRING_get0_data(serialNumber)[a], 16).rightJustified(2, '0');
             hexString += ':';
         }
         hexString.chop(1);
@@ -251,10 +251,10 @@ QMultiMap<QSsl::AlternativeNameEntryType, QString> QSs
                 QHostAddress ipAddress;
                 switch (len) {
                 case 4: // IPv4
-                    ipAddress = QHostAddress(qFromBigEndian(*reinterpret_cast<quint32 *>(genName->d.iPAddress->data)));
+                    ipAddress = QHostAddress(qFromBigEndian(*reinterpret_cast<const quint32 *>(q_ASN1_STRING_get0_data(genName->d.iPAddress))));
                     break;
                 case 16: // IPv6
-                    ipAddress = QHostAddress(reinterpret_cast<quint8 *>(genName->d.iPAddress->data));
+                    ipAddress = QHostAddress(reinterpret_cast<const quint8 *>(q_ASN1_STRING_get0_data(genName->d.iPAddress)));
                     break;
                 default: // Unknown IP address format
                     break;
@@ -511,8 +511,8 @@ static QVariant x509ExtensionToValue(X509_EXTENSION *e
 
             // keyid
             if (auth_key->keyid) {
-                QByteArray keyid(reinterpret_cast<const char *>(auth_key->keyid->data),
-                                 auth_key->keyid->length);
+                QByteArray keyid(reinterpret_cast<const char *>(q_ASN1_STRING_get0_data(auth_key->keyid)),
+                                 q_ASN1_STRING_length(auth_key->keyid));
                 result[QLatin1String("keyid")] = keyid.toHex();
             }
 
